src/quota_aggregator_impl.cc

```diff
@@ -144,43 +146,30 @@ ::google::protobuf::util::Status QuotaAggregatorImpl::CacheResponse(
     return ::google::protobuf::util::Status::OK;
   }
 
+  string request_signature = GenerateAllocateQuotaRequestSignature(request);
+
+  CacheElem* cache_elem = new CacheElem(response, SimpleCycleTimer::Now());
+  cache_elem->set_signature(request_signature);
+
   AllocateQuotaCacheRemovedItemsHandler::StackBuffer stack_buffer(this);
   MutexLock lock(cache_mutex_);
   AllocateQuotaCacheRemovedItemsHandler::StackBuffer::Swapper swapper(
       this, &stack_buffer);
 
-  InternalCacheResponse(request, response, false);
-
-  return ::google::protobuf::util::Status::OK;
-}
-
-// This method must be called behind the cache_mutex_ lock
-void QuotaAggregatorImpl::InternalCacheResponse(
-    const ::google::api::servicecontrol::v1::AllocateQuotaRequest& request,
-    const ::google::api::servicecontrol::v1::AllocateQuotaResponse& response,
-    bool is_refreshing) {
-  if (!cache_) {
-    return;
-  }
-
-  string request_signature = GenerateAllocateQuotaRequestSignature(request);
   QuotaCache::ScopedLookup lookup(cache_.get(), request_signature);
-
-  CacheElem* cache_elem = new CacheElem(response, SimpleCycleTimer::Now());
-  cache_elem->set_signature(request_signature);
-
   if (lookup.Found()) {
     if (lookup.value()->is_aggregated()) {
       // Cached response will be aggregated to the new element for the response
       cache_elem->Aggregate(lookup.value()->ReturnAllocateQuotaRequestAndClear(
```

Comment:
I don't understand this.   what wee need is:

for positive response,  we just drop it.
for negative response, we use it to override the cached item.

I don't see the logic here.